ecs/client.go

```diff
@@ -20,52 +20,53 @@ import (
 // ECS API. It supports retrying requests using exponential backoff and jitter.
 type BasicClient struct {
 	awsutil.BaseClient
-	ecs *ecs.ECS
+	ecs *ecs.Client
 }
 
 // NewBasicClient creates a new AWS ECS client from the given options.
-func NewBasicClient(opts awsutil.ClientOptions) (*BasicClient, error) {
+func NewBasicClient(ctx context.Context, opts awsutil.ClientOptions) (*BasicClient, error) {
 	c := &BasicClient{
 		BaseClient: awsutil.NewBaseClient(opts),
 	}
-	if err := c.setup(); err != nil {
+	if err := c.setup(ctx); err != nil {
 		return nil, errors.Wrap(err, "setting up client")
 	}
 
 	return c, nil
 }
 
-func (c *BasicClient) setup() error {
+func (c *BasicClient) setup(ctx context.Context) error {
 	if c.ecs != nil {
 		return nil
 	}
 
-	sess, err := c.GetSession()
+	config, err := c.GetConfig(ctx)
 	if err != nil {
 		return errors.Wrap(err, "initializing session")
 	}
 
-	c.ecs = ecs.New(sess)
+	c.ecs = ecs.NewFromConfig(*config)
 
 	return nil
 }
 
 // RegisterTaskDefinition registers a new task definition.
 func (c *BasicClient) RegisterTaskDefinition(ctx context.Context, in *ecs.RegisterTaskDefinitionInput) (*ecs.RegisterTaskDefinitionOutput, error) {
-	if err := c.setup(); err != nil {
+	if err := c.setup(ctx); err != nil {
 		return nil, errors.Wrap(err, "setting up client")
 	}
 
 	var out *ecs.RegisterTaskDefinitionOutput
 	var err error
 	if err := utility.Retry(ctx, func() (bool, error) {
 		msg := awsutil.MakeAPILogMessage("RegisterTaskDefinition", in)
-		out, err = c.ecs.RegisterTaskDefinitionWithContext(ctx, in)
-		if awsErr, ok := err.(awserr.Error); ok {
-			grip.Debug(message.WrapError(awsErr, msg))
-			if c.isNonRetryableErrorCode(awsErr.Code()) {
-				return false, err
-			}
+		out, err = c.ecs.RegisterTaskDefinition(ctx, in)
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) {
```

Comment:
Now that I'm staring at this (which was basically copy/pasted from the Evergreen EC2 client implementation), it seems like we can unconditionally log the error message without checking the type conversion. Can we remove all the `errors.As` conversions for the debug logs?

Reply:
done